include/oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h

```diff
@@ -36,24 +38,98 @@ class sycl_backend
     using resource_container_t = std::vector<execution_resource_t>;
 
   private:
-    class async_waiter
+    static inline bool is_profiling_enabled = false;
+    using report_clock_type = std::chrono::steady_clock;
+    using report_duration = std::chrono::milliseconds;
+
+    class async_waiter_base
+    {
+      public:
+        virtual void
+        report() = 0;
+        virtual bool
+        is_complete() = 0;
+        virtual ~async_waiter_base() = default;
+    };
```

Comment:
I propose to skip clang format for this part and rewrite it as
```suggestion
    class async_waiter_base
    {
      public:
        virtual void report() = 0;
        virtual bool is_complete() = 0;
        virtual ~async_waiter_base() = default;
    };
```
because it's interface declaration and should be readable.

Reply:
fixed